cmd/convertor/builder/builder_engine.go

```diff
@@ -149,10 +149,10 @@ func (e *builderEngineBase) mediaTypeImageLayer() string {
 	}
 }
 
-func (e *builderEngineBase) uploadManifestAndConfig(ctx context.Context) error {
+func (e *builderEngineBase) uploadManifestAndConfig(ctx context.Context) (specs.Descriptor, error) {
 	cbuf, err := json.Marshal(e.config)
 	if err != nil {
-		return err
+		return specs.DescriptorEmptyJSON, err
```

Comment:
Nit. The empty descriptor is a valid descriptor used for referrer artifacts. Should use specs.Descriptor{} with the error instead.